src/main/java/org/spongepowered/common/data/manipulator/immutable/entity/ImmutableSpongeBodyPartRotationalData.java

```diff
@@ -0,0 +1,139 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.manipulator.immutable.entity;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.flowpowered.math.vector.Vector3d;
+import com.google.common.collect.ImmutableMap;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.MemoryDataContainer;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableBodyPartRotationalData;
+import org.spongepowered.api.data.manipulator.mutable.entity.BodyPartRotationalData;
+import org.spongepowered.api.data.type.BodyPart;
+import org.spongepowered.api.data.type.BodyParts;
+import org.spongepowered.api.data.value.immutable.ImmutableMapValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.common.data.manipulator.immutable.common.AbstractImmutableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeBodyPartRotationalData;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeMapValue;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+import java.util.Map;
+
+public class ImmutableSpongeBodyPartRotationalData extends AbstractImmutableData<ImmutableBodyPartRotationalData, BodyPartRotationalData> 
+    implements ImmutableBodyPartRotationalData {
+
+    private final Map<BodyPart, Vector3d> rotations;
+
+    public ImmutableSpongeBodyPartRotationalData(Map<BodyPart, Vector3d> rotations) {
+        super(ImmutableBodyPartRotationalData.class);
+        
+        this.rotations = ImmutableMap.copyOf(checkNotNull(rotations));
+    }
+
+    @Override
+    public BodyPartRotationalData asMutable() {
+        return new SpongeBodyPartRotationalData(this.rotations);
+    }
+
+    @Override
+    public int compareTo(ImmutableBodyPartRotationalData o) {
+        return 0;
+    }
+
+    @Override
+    public DataContainer toContainer() {
+        return new MemoryDataContainer()
+                .set(Keys.BODY_ROTATIONS, this.rotations)
+                .set(Keys.HEAD_ROTATION, this.rotations.get(BodyParts.HEAD))
+                .set(Keys.CHEST_ROTATION, this.rotations.get(BodyParts.CHEST))
+                .set(Keys.LEFT_ARM_ROTATION, this.rotations.get(BodyParts.LEFT_ARM))
+                .set(Keys.RIGHT_ARM_ROTATION, this.rotations.get(BodyParts.RIGHT_ARM))
+                .set(Keys.LEFT_LEG_ROTATION, this.rotations.get(BodyParts.LEFT_LEG))
+                .set(Keys.RIGHT_LEG_ROTATION, this.rotations.get(BodyParts.RIGHT_LEG));
+    }
+
+    @Override
+    public ImmutableMapValue<BodyPart, Vector3d> partRotation() {
+        return new ImmutableSpongeMapValue<>(Keys.BODY_ROTATIONS, this.rotations);
```

Comment:
Shouldn't need to create new instances of the immutable value. Store these in a field and create them in the constructor